django/thunderstore/repository/api/v1/tests/test_api_v1.py

```diff
@@ -1,32 +1,128 @@
+    )
     update_api_v1_caches()
+    cache = APIV1PackageCache.get_latest_for_community(active_package_listing.community)
+    assert cache is not None
+
+    # Should get a full response
     response = api_client.get("/api/v1/package/")
     assert response.status_code == 200
-    result = response.json()
+
+    # The response is gzipped
+    content = BytesIO(response.content)
+    with gzip.GzipFile(fileobj=content, mode="r") as f:
+        result = json.loads(f.read())
+
     assert len(result) == 1
     assert result[0]["name"] == active_package_listing.package.name
     assert result[0]["full_name"] == active_package_listing.package.full_package_name
+    last_modified = response["Last-Modified"]
+    assert last_modified == http_date(int(cache.last_modified.timestamp()))
+    assert response["Content-Type"] == cache.content_type
+    assert response["Content-Encoding"] == cache.content_encoding
 
-    # TODO: Enable once detail views have been re-enabled
-    # uuid = result[0]["uuid4"]
-    # response = api_client.get(
-    #     f"/api/v1/package/{uuid}/",
-    # )
-    # assert response.status_code == 200
-    # assert response.json() == result[0]
+    # Should get a 304 since Last-Modified matches
+    response = api_client.get(
+        path="/api/v1/package/",
+        HTTP_IF_MODIFIED_SINCE=last_modified,
+    )
+    assert response.status_code == 304
+
+    # We need to sleep at least 0.5 seconds to ensure differing timestamp
+    # TODO: Use freezegun or similar instead of sleeping
+    # TODO: Use ETag instead of just timestmap
+    time.sleep(1)
+    update_api_v1_caches()
+    new_cache = APIV1PackageCache.get_latest_for_community(
+        active_package_listing.community
+    )
+    assert new_cache != cache
+    assert new_cache.last_modified > cache.last_modified
+
+    # Should get a 200 since cache was regenerated
+    response = api_client.get(
+        path="/api/v1/package/", HTTP_IF_MODIFIED_SINCE=last_modified
+    )
+    assert response.status_code == 200
+    assert response["Last-Modified"] != last_modified
+    assert response["Last-Modified"] == http_date(
+        int(new_cache.last_modified.timestamp())
+    )
+    assert len(result) == 1
+    assert result[0]["name"] == active_package_listing.package.name
+    assert result[0]["full_name"] == active_package_listing.package.full_package_name
+    assert result[0]["package_url"].startswith(
+        f"{settings.PROTOCOL}{community_site.site.domain}"
+    )
+    assert result[0]["versions"][0]["download_url"].startswith(
+        f"{settings.PROTOCOL}{community_site.site.domain}"
+    )
+
+
+@pytest.mark.django_db
+def test_api_v1_package_detail(
+    api_client: APIClient,
+    community_site: CommunitySite,
+    active_package_listing: PackageListing,
+) -> None:
+    response = api_client.get(
+        f"/api/v1/package/{active_package_listing.package.uuid4}/",
+    )
+    assert community_site.community == active_package_listing.community
```

Comment:
This could be asserted before making the request, maybe speed up the tests a tiny fraction.